### PR TITLE
Fix vendors enumeration on 32-bit Windows OS

### PR DESCRIPTION
**Pull request by ofirc**
When running 32-bit OpenCL applications on a 32-bit OS, we need to
use the registry keys without the "Wow" suffix. On 64-bit OSes and
32-bit applications, OTOH, we must use the Wow suffix.

### CODE REVIEW

File: icd_windows_hkr.c [hunk 1]
```diff
@@ -63,11 +63,30 @@ typedef enum
         mem = NULL;                 \
     } while (0)
 
-#ifdef _WIN64
 static const char OPENCL_REG_SUB_KEY[] = "OpenCLDriverName";
+
+#ifndef _WIN64
```
Comment (kepatil) [inline icd_windows_hkr.c]: This should be ifdef instead of ifndef, right?

Same in the function below.

**Comment by ofirc** [comment]
@kepatil Any updates :)?

**Comment by kepatil** [comment]
Sorry, I did not notice this comment. @ofirc, can you address the comment I made on your change?

**Comment by ofirc** [comment]
I already addressed them above:
1) ifndef
   The SUB_KEY, OpenCLDriverNameWow, is relevant only to 32-builds so this #ifndef takes care to avoid defining this variable from 64-bit builds (which results in a smaller footprint of OpenCL.dll).

2) ifdef
On 64-bit builds of OpenCL.dll, OTOH, the only valid name is the OPENCL_REG_SUB_KEY, i.e. OpenCLDriverName, so we return earliest as possible (as no checks are needed).

Everything that can be deferred to compile time is preferred, and every byte that is not needed is saved.

**Comment by kepatil** [comment]
Okay. For some reason I your comments on the change were not visible to me till you posted them as a separate comment. The change looks good to me after your explanation. I had gotten the logic backwards in my mind. I'll merge this.

**Pull request merged**